lib/srv/ctx.go

```diff
@@ -676,24 +678,61 @@ func (c *ServerContext) String() string {
 	return fmt.Sprintf("ServerContext(%v->%v, user=%v, id=%v)", c.ServerConn.RemoteAddr(), c.ServerConn.LocalAddr(), c.ServerConn.User(), c.id)
 }
 
-// ExecCommand takes a *ServerContext and extracts the parts needed to create
-// an *execCommand which can be re-sent to Teleport.
-func (c *ServerContext) ExecCommand() (*ExecCommand, error) {
-	var pamEnabled bool
-	var pamServiceName string
-	var pamUseAuth bool
+func getPAMConfig(c *ServerContext) (*PAMConfig, error) {
+	// PAM should be disabled.
+	if c.srv.Component() != teleport.ComponentNode {
+		return nil, nil
+	}
```

Comment:
Still curious what the error is, this shouldn't be needed.

Reply:
Hmm, not sure if anyone modified integration tests recently but the error is gone after rebasing on latest master.

Reply:
May have been a faulty test if so.

Reply:
Nope, ran into it again, it's at `ctx.go:688` on this branch and stems from tests not passing a full fileconf. The one passed does not include PAM and the component field is set differently to keep PAM from triggering. This seems to b happening in a few different tests.

Reply:
The fileconf passed results in `c.srv.GetPAM()` returning `nil, nil` which should normally be impossible but the test does not fill in all configuration values.

Reply:
Seems to stem from `FakeServer` not filling out configuration.

Reply:
We have two options here, add back in the check as it exists currently on master or refactor tests and `FakeServer`. Option b feels out of scope here.